src/plugins/query_enhancements/server/search/sql_async_search_strategy.ts

```diff
@@ -44,45 +38,61 @@ export const sqlAsyncSearchStrategyProvider = (
     search: async (context, request: any, options) => {
       try {
         const query: Query = request.body.query;
-        const startTime = Date.now();
-        request.body = { ...request.body, lang: SEARCH_STRATEGY.SQL };
-        const rawResponse: any = await sqlAsyncFacet.describeQuery(context, request);
+        const pollQueryResultsParams = request.body.pollQueryResultsParams;
+        const inProgressQueryId = pollQueryResultsParams?.queryId;
 
-        if (!rawResponse.success) handleFacetError(rawResponse);
+        if (!inProgressQueryId) {
+          request.body = { ...request.body, lang: SEARCH_STRATEGY.SQL };
+          const rawResponse: any = await sqlAsyncFacet.describeQuery(context, request);
 
-        const statusConfig = buildQueryStatusConfig(rawResponse);
-        request.params = { queryId: statusConfig.queryId };
+          if (!rawResponse.success) handleFacetError(rawResponse);
 
-        const response = await handleQueryStatus({
-          fetchStatus: async () => {
-            const status: any = await sqlAsyncJobsFacet.describeQuery(context, request);
-            logger.info(
-              `sqlAsyncSearchStrategy: JOB: ${statusConfig.queryId} - STATUS: ${status.data?.status}`
-            );
-            return status;
-          },
-          isServer: true,
-        });
+          const statusConfig = buildQueryStatusConfig(rawResponse);
 
-        const dataFrame = createDataFrame({
-          name: query.dataset?.id,
-          schema: response.data.schema,
-          meta: statusConfig,
-          fields: getFields(response),
-        });
+          return {
+            type: DATA_FRAME_TYPES.POLLING,
+            status: 'started',
+            body: {
+              queryStatusConfig: statusConfig,
+            },
+          } as IDataFrameResponse;
+        } else {
+          request.params = { queryId: inProgressQueryId };
+          const queryStatusResponse: any = await sqlAsyncJobsFacet.describeQuery(context, request);
+          const queryStatus = queryStatusResponse?.data?.status;
+          logger.info(`sqlAsyncSearchStrategy: JOB: ${inProgressQueryId} - STATUS: ${queryStatus}`);
 
-        const elapsedMs = Date.now() - startTime;
+          if (queryStatus?.toUpperCase() === 'SUCCESS') {
+            const dataFrame = createDataFrame({
+              name: query.dataset?.id,
+              schema: queryStatusResponse.data.schema,
+              meta: { ...pollQueryResultsParams },
+              fields: getFields(queryStatusResponse),
+            });
 
-        dataFrame.size = response.data.datarows.length;
+            dataFrame.size = queryStatusResponse.data.datarows.length;
```

Comment:
Shouldn't we check that `datarows` is available and an array?

Reply:
I will leave that decision to the owners of this strategy, this PR is only updating the strategy to use the polling mechanism

Reply:
Will follow up on this with the right owners and make change if needed in follow up PR